sdk/src/main/scala/com/horizen/account/state/EoaMessageProcessor.scala

```diff
@@ -0,0 +1,44 @@
+package com.horizen.account.state
+
+import scorex.util.ScorexLogging
+
+import java.math.BigInteger
+import scala.util.Failure
+
+/*
+ * EoaMessageProcessor is responsible for management of regular coin transfers inside sidechain.
+ * In our case to make a transfer from one user account (EOA account) to another user account.
+ */
+object EoaMessageProcessor extends MessageProcessor with ScorexLogging {
+  //TODO: actual gas to be defined
+  val gasUsed: BigInteger = BigInteger.valueOf(20000L)
+
+  override def init(view: AccountStateView): Unit = {
+    // No actions required for transferring coins during genesis state initialization.
+  }
+
+  override def canProcess(msg: Message, view: AccountStateView): Boolean = {
+    // Can process only EOA to EOA transfer, so when:
+    // 1. "from" account must be an EOA account - no "code" set;
+    // 2. transaction "data" is empty.
+    msg.getData.isEmpty && view.getCodeHash(msg.getFrom.address()).isEmpty
+  }
+
+  override def process(msg: Message, view: AccountStateView): ExecutionResult = {
+    view.subBalance(msg.getFrom.address(), msg.getValue) match {
```

Comment:
I assume we have already validated that the sender has enough balance for the `value` and gas fees, correct?
So far `subBalance` does not do any validation and will also happily set the accounts balance to a negative value.

Reply:
right, it is checked outside in AccountState, and theoretically never can occur. But considering that interface allow it to fail -> better to process this case.